analysis.tex

```diff
@@ -1,29 +1,53 @@
 \section{Analysis}
 
-\begin{theorem}
-  If $H$ is modelled as a \emph{Random Oracle}, then the protocol $\Pi$ of Section~\ref{section:construction} is \emph{unspendable}.
+\begin{theorem}[Correctness]
+  The Proof-of-Burn protocol $\Pi$ of Section~\ref{section:construction} is \emph{correct}.
 \end{theorem}
 \begin{proof}
-  $H(t) \xor 1 = pkh$.
-  Adversary knows a $pk$ st $H(pk) = pkh$.
-  This is impossible under the Random Oracle model.
-  TODO
+  We need to prove that $\forall \kappa, t. \BurnVerify(1^\kappa, t, \GenBurnAddress(1^\kappa, t)) = \textsf{true}$.
+
+  Based on Algorithm~\ref{alg.construction}, $\BurnVerify(1^\kappa, t, \GenBurnAddress(1^\kappa, t)) = \textsf{true}$ if and only if $\GenBurnAddress(1^\kappa, t) = \GenBurnAddress(1^\kappa, t)$, which always holds as $\GenBurnAddress$ is deterministic.
+\end{proof}
+
+\begin{theorem}[Unspendability]
+  If $H$ is modelled as a preimage-resistant hash function, then the protocol $\Pi$ of Section~\ref{section:construction} is \emph{unspendable}.
+\end{theorem}
+\begin{proof}
+  Assume that the protocol $\Pi$ is not unspendable. This means that there exists a probabilistic polynomial-time adversary $\mathcal{A}$ such that:
```

Comment:
I'd prefer we shorten our proofs so that they are not by contradiction and instead are more constructive in nature (and closer to intuitionism in that we're not forced to use the law of excluded middle needlessly). Instead, use Katz's constructive style of proofs. The theorem states that if H is hard, then A does not win except with negligible probability. Then the proof works _without assuming that A wins with non-negligible probability_. First create an adversary A* against H which uses A. Then show that the probability of A* succeeding is bounds the probability of A succeeding (if A succeeds, then A* succeeds). Therefore Pr[A* success] >= Pr[A success]. From the assumption that Pr[A* success] is negl, therefore Pr[A success] is negligible also.